source/credentials.c

```diff
@@ -213,34 +230,82 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
         providers[index++] = profile_provider;
     }
 
-    struct aws_credentials_provider_imds_options imds_options;
-    AWS_ZERO_STRUCT(imds_options);
-    imds_options.bootstrap = options->bootstrap;
-    imds_provider = aws_credentials_provider_new_imds(allocator, &imds_options);
-    if (imds_provider == NULL) {
-        goto on_error;
-    }
+    if (ecs_relative_uri && ecs_relative_uri->len) {
+        struct aws_credentials_provider_ecs_options ecs_options;
+        AWS_ZERO_STRUCT(ecs_options);
+        ecs_options.bootstrap = options->bootstrap;
+        ecs_options.host = aws_byte_cursor_from_string(s_ecs_host);
+        ecs_options.path_and_query = aws_byte_cursor_from_string(ecs_relative_uri);
+        ecs_options.use_tls = false;
```

Comment:
Don't change this, you're fitting in with the style of the code around you. I just wanted to share, in case you didn't know, we can do pretty things like this in C:
```
struct aws_credentials_provider_ecs_options ecs_options = {
    .bootstrap = options->bootstrap,
    .host = aws_byte_cursor_from_string(s_ecs_host),
    .path_and_query = aws_byte_cursor_from_string(ecs_relative_uri),
    .use_tls = false,
};
```